tests/components/control4/test_config_flow.py

```diff
@@ -0,0 +1,105 @@
+"""Test the Control4 config flow."""
+from homeassistant import config_entries, setup
+from homeassistant.components.control4.const import DOMAIN
+
+from tests.async_mock import patch, AsyncMock
+import datetime
+from pyControl4.error_handling import Unauthorized
+
+
+async def test_form(hass):
+    """Test we get the form."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["errors"] == {}
+
+    with patch(
+        "pyControl4.account.C4Account.getAccountControllers",
```

Comment:
@bdraco I've added the return values expected from the pyControl4 package, but it doesn't seem to actually work (the variables that are assigned these values are returning as `None` when I run the test)

Could you take a look and suggest how this is supposed to be done?

Reply:
You need to mock the `pyControl4.account.C4Account` objection creation with a pytest fixture or something like `harmony/test_config_flow.py` does

Reply:
After looking at `harmony/test_config_flow.py`, I've gotten `test_form` to work. 
However, I can't get `test_form_invalid_auth` and `test_form_cannot_connect` to pass. Even though I'm raising the same `Unauthorized` exceptions as a `side_effect`, they don't seem to be caught properly by `config_flow`.

Reply:
This should fix it https://github.com/home-assistant/core/pull/37632#discussion_r454672497